tests/workflows/test_main_snapshot.py

```
@@ -0,0 +1,185 @@
+
+        assert result == 1, f"expected exit code 1, got {result}"
+
+    def test_main_missing_file_raises(self, tmp_path: Path) -> None:
+        """main raises an exception for a non-existent manifest file."""
+        missing = tmp_path / "does_not_exist.toml"
+
+        with patch(
+            "sys.argv",
+            ["dependency_binaries_manifest.py", str(missing)],
+        ):
+            with pytest.raises(FileNotFoundError, match=""):
```

Ruff (0.15.9)

[warning] Use a single `with` statement with multiple contexts instead of nested `with` statements

Combine `with` statements

(SIM117)